portfolio/src/main/webapp/script.js

```diff
@@ -35,27 +35,60 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+/**
+ * Updates the sort order and number of comments to display in the server,
+ * and fetches them according to the new parameters.
+ */
+function updateMessage() {
+  let headers = new Headers();
+  headers.append('numComments', getParameter('numComments'));
+  headers.append('commentSort', getParameter('commentSort'));
+  headers.append('change', true);
+
+  fetch('/data', {headers: headers}).then(response => response.json()).then((messages) => 
+      getMessages(messages));
+}
+
 /**
  * Fetches the message from the servlet and displays it on the page.
  */
-function getMessage() {
-  fetch('/data').then(response => response.json()).then((messages) => {
+function initMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => getMessages(messages));
+}
+
+/**
+ * Insert all messages from the given list into a new div on the page
+ */
+function getMessages(messages) {
     console.log(messages);
+
+    let section = document.getElementById('display-comments');
     let commentsDiv = document.createElement('div');
-    let numDisplay = getParameter("numComments");
-    if (numDisplay == null) {
-      numDisplay = 10;  // Default value.
-    }
-    for (index = 0; index < messages.length && index < numDisplay; index++) {
-      let comment = document.createElement('p');
-      let linebreak = document.createElement('br');
-      comment.textContent = messages[index];
 
-      commentsDiv.appendChild(comment);
-      commentsDiv.appendChild(linebreak);
+    let sortNum = messages[messages.length - 1];
+    let colonIndex = sortNum.indexOf(";");
```

Comment:
[nit] Probably should be called 'semicolonIndex' since the character ';' is a semicolon, not a colon.